Several warnings and lots of ugly output when installing from npm

This library is an indirect dependency in our project and spews a bunch of warnings and a lot of gross output when it's installed. Is there a way to fix this?

```
20:05:47  .../node_modules/tree-sitter-json install$ node-gyp rebuild
20:05:47  .../esbuild@0.18.10/node_modules/esbuild postinstall$ node install.js
20:05:47  .../node_modules/tree-sitter-yaml install$ node-gyp rebuild
20:05:47  .../esbuild@0.18.10/node_modules/esbuild postinstall: Done
20:05:47  .../node_modules/tree-sitter-json install: gyp info it worked if it ends with ok
20:05:47  .../node_modules/tree-sitter-json install: gyp info using node-gyp@9.4.0
20:05:47  .../node_modules/tree-sitter-json install: gyp info using node@18.16.0 | linux | x64
20:05:47  .../node_modules/tree-sitter-yaml install: gyp info it worked if it ends with ok
20:05:47  .../node_modules/tree-sitter-yaml install: gyp info using node-gyp@9.4.0
20:05:47  .../node_modules/tree-sitter-yaml install: gyp info using node@18.16.0 | linux | x64
20:05:47  .../node_modules/tree-sitter-json install: gyp info find Python using Python version 3.11.2 found at "/usr/bin/python3"
20:05:47  .../node_modules/tree-sitter-yaml install: gyp info find Python using Python version 3.11.2 found at "/usr/bin/python3"
20:05:47  .../node_modules/tree-sitter-json install: gyp http GET https://nodejs.org/download/release/v18.16.0/node-v18.16.0-headers.tar.gz
20:05:47  .../node_modules/tree-sitter-yaml install: gyp http GET https://nodejs.org/download/release/v18.16.0/node-v18.16.0-headers.tar.gz
20:05:47  .../node_modules/tree-sitter-json install: gyp http 200 https://nodejs.org/download/release/v18.16.0/node-v18.16.0-headers.tar.gz
20:05:47  .../node_modules/tree-sitter-yaml install: gyp http 200 https://nodejs.org/download/release/v18.16.0/node-v18.16.0-headers.tar.gz
20:05:49  .../node_modules/tree-sitter-json install: gyp http GET https://nodejs.org/download/release/v18.16.0/SHASUMS256.txt
20:05:49  .../node_modules/tree-sitter-yaml install: gyp http GET https://nodejs.org/download/release/v18.16.0/SHASUMS256.txt
20:05:49  .../node_modules/tree-sitter-yaml install: gyp http 200 https://nodejs.org/download/release/v18.16.0/SHASUMS256.txt
20:05:49  .../node_modules/tree-sitter-json install: gyp http 200 https://nodejs.org/download/release/v18.16.0/SHASUMS256.txt
20:05:49  .../node_modules/tree-sitter-yaml install: gyp info spawn /usr/bin/python3
20:05:49  .../node_modules/tree-sitter-yaml install: gyp info spawn args [
20:05:49  .../node_modules/tree-sitter-yaml install: gyp info spawn args   '/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
20:05:49  .../node_modules/tree-sitter-yaml install: gyp info spawn args   'binding.gyp',
20:05:49  .../node_modules/tree-sitter-yaml install: gyp info spawn args   '-f',
20:05:49  .../node_modules/tree-sitter-yaml install: gyp info spawn args   'make',
20:05:49  .../node_modules/tree-sitter-yaml install: gyp info spawn args   '-I',
20:05:49  .../node_modules/tree-sitter-yaml install: gyp info spawn args   '/home/jenkins/workspace/Geiger/geiger-review/node_modules/.pnpm/tree-sitter-yaml@0.5.0/node_modules/tree-sitter-yaml/build/config.gypi',
20:05:49  .../node_modules/tree-sitter-yaml install: gyp info spawn args   '-I',
20:05:49  .../node_modules/tree-sitter-yaml install: gyp info spawn args   '/usr/local/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
20:05:49  .../node_modules/tree-sitter-yaml install: gyp info spawn args   '-I',
20:05:49  .../node_modules/tree-sitter-yaml install: gyp info spawn args   '/home/jenkins/.cache/node-gyp/18.16.0/include/node/common.gypi',
20:05:49  .../node_modules/tree-sitter-yaml install: gyp info spawn args   '-Dlibrary=shared_library',
20:05:49  .../node_modules/tree-sitter-yaml install: gyp info spawn args   '-Dvisibility=default',
20:05:49  .../node_modules/tree-sitter-yaml install: gyp info spawn args   '-Dnode_root_dir=/home/jenkins/.cache/node-gyp/18.16.0',
20:05:49  .../node_modules/tree-sitter-yaml install: gyp info spawn args   '-Dnode_gyp_dir=/usr/local/lib/node_modules/npm/node_modules/node-gyp',
20:05:49  .../node_modules/tree-sitter-yaml install: gyp info spawn args   '-Dnode_lib_file=/home/jenkins/.cache/node-gyp/18.16.0/<(target_arch)/node.lib',
20:05:49  .../node_modules/tree-sitter-yaml install: gyp info spawn args   '-Dmodule_root_dir=/home/jenkins/workspace/Geiger/geiger-review/node_modules/.pnpm/tree-sitter-yaml@0.5.0/node_modules/tree-sitter-yaml',
20:05:49  .../node_modules/tree-sitter-yaml install: gyp info spawn args   '-Dnode_engine=v8',
20:05:49  .../node_modules/tree-sitter-yaml install: gyp info spawn args   '--depth=.',
20:05:49  .../node_modules/tree-sitter-yaml install: gyp info spawn args   '--no-parallel',
20:05:49  .../node_modules/tree-sitter-yaml install: gyp info spawn args   '--generator-output',
20:05:49  .../node_modules/tree-sitter-yaml install: gyp info spawn args   'build',
20:05:49  .../node_modules/tree-sitter-yaml install: gyp info spawn args   '-Goutput_dir=.'
20:05:49  .../node_modules/tree-sitter-yaml install: gyp info spawn args ]
20:05:49  .../node_modules/tree-sitter-json install: gyp info spawn /usr/bin/python3
20:05:49  .../node_modules/tree-sitter-json install: gyp info spawn args [
20:05:49  .../node_modules/tree-sitter-json install: gyp info spawn args   '/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
20:05:49  .../node_modules/tree-sitter-json install: gyp info spawn args   'binding.gyp',
20:05:49  .../node_modules/tree-sitter-json install: gyp info spawn args   '-f',
20:05:49  .../node_modules/tree-sitter-json install: gyp info spawn args   'make',
20:05:49  .../node_modules/tree-sitter-json install: gyp info spawn args   '-I',
20:05:49  .../node_modules/tree-sitter-json install: gyp info spawn args   '/home/jenkins/workspace/Geiger/geiger-review/node_modules/.pnpm/tree-sitter-json@0.20.0/node_modules/tree-sitter-json/build/config.gypi',
20:05:49  .../node_modules/tree-sitter-json install: gyp info spawn args   '-I',
20:05:49  .../node_modules/tree-sitter-json install: gyp info spawn args   '/usr/local/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
20:05:49  .../node_modules/tree-sitter-json install: gyp info spawn args   '-I',
20:05:49  .../node_modules/tree-sitter-json install: gyp info spawn args   '/home/jenkins/.cache/node-gyp/18.16.0/include/node/common.gypi',
20:05:49  .../node_modules/tree-sitter-json install: gyp info spawn args   '-Dlibrary=shared_library',
20:05:49  .../node_modules/tree-sitter-json install: gyp info spawn args   '-Dvisibility=default',
20:05:49  .../node_modules/tree-sitter-json install: gyp info spawn args   '-Dnode_root_dir=/home/jenkins/.cache/node-gyp/18.16.0',
20:05:49  .../node_modules/tree-sitter-json install: gyp info spawn args   '-Dnode_gyp_dir=/usr/local/lib/node_modules/npm/node_modules/node-gyp',
20:05:49  .../node_modules/tree-sitter-json install: gyp info spawn args   '-Dnode_lib_file=/home/jenkins/.cache/node-gyp/18.16.0/<(target_arch)/node.lib',
20:05:49  .../node_modules/tree-sitter-json install: gyp info spawn args   '-Dmodule_root_dir=/home/jenkins/workspace/Geiger/geiger-review/node_modules/.pnpm/tree-sitter-json@0.20.0/node_modules/tree-sitter-json',
20:05:49  .../node_modules/tree-sitter-json install: gyp info spawn args   '-Dnode_engine=v8',
20:05:49  .../node_modules/tree-sitter-json install: gyp info spawn args   '--depth=.',
20:05:49  .../node_modules/tree-sitter-json install: gyp info spawn args   '--no-parallel',
20:05:49  .../node_modules/tree-sitter-json install: gyp info spawn args   '--generator-output',
20:05:49  .../node_modules/tree-sitter-json install: gyp info spawn args   'build',
20:05:49  .../node_modules/tree-sitter-json install: gyp info spawn args   '-Goutput_dir=.'
20:05:49  .../node_modules/tree-sitter-json install: gyp info spawn args ]
20:05:49  .../node_modules/tree-sitter-json install: gyp info spawn make
20:05:49  .../node_modules/tree-sitter-json install: gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
20:05:49  .../node_modules/tree-sitter-json install: make: Entering directory '/home/jenkins/workspace/Geiger/geiger-review/node_modules/.pnpm/tree-sitter-json@0.20.0/node_modules/tree-sitter-json/build'
20:05:49  .../node_modules/tree-sitter-json install:   CC(target) Release/obj.target/tree_sitter_json_binding/src/parser.o
20:05:49  .../node_modules/tree-sitter-yaml install: gyp info spawn make
20:05:49  .../node_modules/tree-sitter-yaml install: gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
20:05:49  .../node_modules/tree-sitter-yaml install: make: Entering directory '/home/jenkins/workspace/Geiger/geiger-review/node_modules/.pnpm/tree-sitter-yaml@0.5.0/node_modules/tree-sitter-yaml/build'
20:05:49  .../node_modules/tree-sitter-yaml install:   CXX(target) Release/obj.target/tree_sitter_yaml_binding/bindings/node/binding.o
20:05:49  .../node_modules/tree-sitter-json install:   CXX(target) Release/obj.target/tree_sitter_json_binding/bindings/node/binding.o
20:05:50  .../node_modules/tree-sitter-yaml install: In file included from ../bindings/node/binding.cc:2:
20:05:50  .../node_modules/tree-sitter-yaml install: /home/jenkins/.cache/node-gyp/18.16.0/include/node/node.h:997:7: warning: cast between incompatible function types from 'void (*)(v8::Local<v8::Object>, v8::Local<v8::Object>)' to 'node::addon_register_func' {aka 'void (*)(v8::Local<v8::Object>, v8::Local<v8::Value>, void*)'} [-Wcast-function-type]
20:05:50  .../node_modules/tree-sitter-yaml install:   997 |       (node::addon_register_func) (regfunc),                          \
20:05:50  .../node_modules/tree-sitter-yaml install:       |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
20:05:50  .../node_modules/tree-sitter-yaml install: /home/jenkins/.cache/node-gyp/18.16.0/include/node/node.h:1031:3: note: in expansion of macro 'NODE_MODULE_X'
20:05:50  .../node_modules/tree-sitter-yaml install:  1031 |   NODE_MODULE_X(modname, regfunc, NULL, 0)  // NOLINT (readability/null_usage)
20:05:50  .../node_modules/tree-sitter-yaml install:       |   ^~~~~~~~~~~~~
20:05:50  .../node_modules/tree-sitter-yaml install: ../bindings/node/binding.cc:26:1: note: in expansion of macro 'NODE_MODULE'
20:05:50  .../node_modules/tree-sitter-yaml install:    26 | NODE_MODULE(tree_sitter_yaml_binding, Init)
20:05:50  .../node_modules/tree-sitter-yaml install:       | ^~~~~~~~~~~
20:05:50  .../node_modules/tree-sitter-json install: In file included from ../bindings/node/binding.cc:2:
20:05:50  .../node_modules/tree-sitter-json install: /home/jenkins/.cache/node-gyp/18.16.0/include/node/node.h:997:7: warning: cast between incompatible function types from 'void (*)(v8::Local<v8::Object>, v8::Local<v8::Object>)' to 'node::addon_register_func' {aka 'void (*)(v8::Local<v8::Object>, v8::Local<v8::Value>, void*)'} [-Wcast-function-type]
20:05:50  .../node_modules/tree-sitter-json install:   997 |       (node::addon_register_func) (regfunc),                          \
20:05:50  .../node_modules/tree-sitter-json install:       |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
20:05:50  .../node_modules/tree-sitter-json install: /home/jenkins/.cache/node-gyp/18.16.0/include/node/node.h:1031:3: note: in expansion of macro 'NODE_MODULE_X'
20:05:50  .../node_modules/tree-sitter-json install:  1031 |   NODE_MODULE_X(modname, regfunc, NULL, 0)  // NOLINT (readability/null_usage)
20:05:50  .../node_modules/tree-sitter-json install:       |   ^~~~~~~~~~~~~
20:05:50  .../node_modules/tree-sitter-json install: ../bindings/node/binding.cc:26:1: note: in expansion of macro 'NODE_MODULE'
20:05:50  .../node_modules/tree-sitter-json install:    26 | NODE_MODULE(tree_sitter_json_binding, Init)
20:05:50  .../node_modules/tree-sitter-json install:       | ^~~~~~~~~~~
20:05:50  .../node_modules/tree-sitter-yaml install:   CC(target) Release/obj.target/tree_sitter_yaml_binding/src/parser.o
20:05:50  .../node_modules/tree-sitter-yaml install: In file included from ../src/parser.c:1:
20:05:50  .../node_modules/tree-sitter-yaml install: ../src/parser.c: In function 'ts_lex':
20:05:50  .../node_modules/tree-sitter-yaml install: ../src/tree_sitter/parser.h:136:11: warning: variable 'lookahead' set but not used [-Wunused-but-set-variable]
20:05:50  .../node_modules/tree-sitter-yaml install:   136 |   int32_t lookahead;            \
20:05:50  .../node_modules/tree-sitter-yaml install:       |           ^~~~~~~~~
20:05:50  .../node_modules/tree-sitter-yaml install: ../src/parser.c:2182:3: note: in expansion of macro 'START_LEXER'
20:05:50  .../node_modules/tree-sitter-yaml install:  2182 |   START_LEXER();
20:05:50  .../node_modules/tree-sitter-yaml install:       |   ^~~~~~~~~~~
20:05:50  .../node_modules/tree-sitter-json install:   SOLINK_MODULE(target) Release/obj.target/tree_sitter_json_binding.node
20:05:50  .../node_modules/tree-sitter-json install:   COPY Release/tree_sitter_json_binding.node
20:05:50  .../node_modules/tree-sitter-json install: make: Leaving directory '/home/jenkins/workspace/Geiger/geiger-review/node_modules/.pnpm/tree-sitter-json@0.20.0/node_modules/tree-sitter-json/build'
20:05:50  .../node_modules/tree-sitter-json install: gyp info ok 
20:05:51  .../node_modules/tree-sitter-json install: Done
20:05:51  .../node_modules/tree-sitter-yaml install:   CXX(target) Release/obj.target/tree_sitter_yaml_binding/src/scanner.o
20:05:51  .../node_modules/tree-sitter-yaml install: ../src/scanner.cc: In member function 'char {anonymous}::Scanner::scn_pln_cnt(TSLexer*, bool ({anonymous}::Scanner::*)(int32_t))':
20:05:51  .../node_modules/tree-sitter-yaml install: ../src/scanner.cc:619:10: warning: variable 'is_cur_wsp' set but not used [-Wunused-but-set-variable]
20:05:51  .../node_modules/tree-sitter-yaml install:   619 |     bool is_cur_wsp = is_wsp(cur_chr);
20:05:51  .../node_modules/tree-sitter-yaml install:       |          ^~~~~~~~~~
20:05:51  .../node_modules/tree-sitter-yaml install: ../src/scanner.cc: In member function 'bool {anonymous}::Scanner::scn_dir_tag_pfx(TSLexer*, TSSymbol)':
20:05:51  .../node_modules/tree-sitter-yaml install: ../src/scanner.cc:66:26: warning: this statement may fall through [-Wimplicit-fallthrough=]
20:05:51  .../node_modules/tree-sitter-yaml install:    66 | #define MRK_END() mrk_end(lexer)
20:05:51  .../node_modules/tree-sitter-yaml install:       |                   ~~~~~~~^~~~~~~
20:05:51  .../node_modules/tree-sitter-yaml install: ../src/scanner.cc:429:11: note: in expansion of macro 'MRK_END'
20:05:51  .../node_modules/tree-sitter-yaml install:   429 |           MRK_END();
20:05:51  .../node_modules/tree-sitter-yaml install:       |           ^~~~~~~
20:05:51  .../node_modules/tree-sitter-yaml install: ../src/scanner.cc:430:9: note: here
20:05:51  .../node_modules/tree-sitter-yaml install:   430 |         case SCN_FAIL:
20:05:51  .../node_modules/tree-sitter-yaml install:       |         ^~~~
20:05:51  .../node_modules/tree-sitter-yaml install: ../src/scanner.cc: In member function 'bool {anonymous}::Scanner::scn_tag(TSLexer*, TSSymbol)':
20:05:51  .../node_modules/tree-sitter-yaml install: ../src/scanner.cc:454:13: warning: this statement may fall through [-Wimplicit-fallthrough=]
20:05:51  .../node_modules/tree-sitter-yaml install:   454 |             if (LKA == '>') {
20:05:51  .../node_modules/tree-sitter-yaml install:       |             ^~
20:05:51  .../node_modules/tree-sitter-yaml install: ../src/scanner.cc:459:11: note: here
20:05:51  .../node_modules/tree-sitter-yaml install:   459 |           case SCN_FAIL:
20:05:51  .../node_modules/tree-sitter-yaml install:       |           ^~~~
20:05:51  .../node_modules/tree-sitter-yaml install: ../src/scanner.cc:66:26: warning: this statement may fall through [-Wimplicit-fallthrough=]
20:05:51  .../node_modules/tree-sitter-yaml install:    66 | #define MRK_END() mrk_end(lexer)
20:05:51  .../node_modules/tree-sitter-yaml install:       |                   ~~~~~~~^~~~~~~
20:05:51  .../node_modules/tree-sitter-yaml install: ../src/scanner.cc:468:13: note: in expansion of macro 'MRK_END'
20:05:51  .../node_modules/tree-sitter-yaml install:   468 |             MRK_END();
20:05:51  .../node_modules/tree-sitter-yaml install:       |             ^~~~~~~
20:05:51  .../node_modules/tree-sitter-yaml install: ../src/scanner.cc:469:11: note: here
20:05:51  .../node_modules/tree-sitter-yaml install:   469 |           case SCN_FAIL:
20:05:51  .../node_modules/tree-sitter-yaml install:       |           ^~~~
20:05:52  .../node_modules/tree-sitter-yaml install:   SOLINK_MODULE(target) Release/obj.target/tree_sitter_yaml_binding.node
20:05:52  .../node_modules/tree-sitter-yaml install:   COPY Release/tree_sitter_yaml_binding.node
20:05:52  .../node_modules/tree-sitter-yaml install: make: Leaving directory '/home/jenkins/workspace/Geiger/geiger-review/node_modules/.pnpm/tree-sitter-yaml@0.5.0/node_modules/tree-sitter-yaml/build'
20:05:52  .../node_modules/tree-sitter-yaml install: gyp info ok 
20:05:52  .../node_modules/tree-sitter-yaml install: Done
20:05:52  .../node_modules/tree-sitter install$ prebuild-install || node-gyp rebuild
20:05:52  .../node_modules/tree-sitter install: prebuild-install warn install No prebuilt binaries found (target=18.16.0 runtime=node arch=x64 libc= platform=linux)
20:05:52  .../node_modules/tree-sitter install: gyp info it worked if it ends with ok
20:05:52  .../node_modules/tree-sitter install: gyp info using node-gyp@9.4.0
20:05:52  .../node_modules/tree-sitter install: gyp info using node@18.16.0 | linux | x64
20:05:52  .../node_modules/tree-sitter install: gyp info find Python using Python version 3.11.2 found at "/usr/bin/python3"
20:05:53  .../node_modules/tree-sitter install: gyp info spawn /usr/bin/python3
20:05:53  .../node_modules/tree-sitter install: gyp info spawn args [
20:05:53  .../node_modules/tree-sitter install: gyp info spawn args   '/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
20:05:53  .../node_modules/tree-sitter install: gyp info spawn args   'binding.gyp',
20:05:53  .../node_modules/tree-sitter install: gyp info spawn args   '-f',
20:05:53  .../node_modules/tree-sitter install: gyp info spawn args   'make',
20:05:53  .../node_modules/tree-sitter install: gyp info spawn args   '-I',
20:05:53  .../node_modules/tree-sitter install: gyp info spawn args   '/home/jenkins/workspace/Geiger/geiger-review/node_modules/.pnpm/tree-sitter@0.20.4/node_modules/tree-sitter/build/config.gypi',
20:05:53  .../node_modules/tree-sitter install: gyp info spawn args   '-I',
20:05:53  .../node_modules/tree-sitter install: gyp info spawn args   '/usr/local/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
20:05:53  .../node_modules/tree-sitter install: gyp info spawn args   '-I',
20:05:53  .../node_modules/tree-sitter install: gyp info spawn args   '/home/jenkins/.cache/node-gyp/18.16.0/include/node/common.gypi',
20:05:53  .../node_modules/tree-sitter install: gyp info spawn args   '-Dlibrary=shared_library',
20:05:53  .../node_modules/tree-sitter install: gyp info spawn args   '-Dvisibility=default',
20:05:53  .../node_modules/tree-sitter install: gyp info spawn args   '-Dnode_root_dir=/home/jenkins/.cache/node-gyp/18.16.0',
20:05:53  .../node_modules/tree-sitter install: gyp info spawn args   '-Dnode_gyp_dir=/usr/local/lib/node_modules/npm/node_modules/node-gyp',
20:05:53  .../node_modules/tree-sitter install: gyp info spawn args   '-Dnode_lib_file=/home/jenkins/.cache/node-gyp/18.16.0/<(target_arch)/node.lib',
20:05:53  .../node_modules/tree-sitter install: gyp info spawn args   '-Dmodule_root_dir=/home/jenkins/workspace/Geiger/geiger-review/node_modules/.pnpm/tree-sitter@0.20.4/node_modules/tree-sitter',
20:05:53  .../node_modules/tree-sitter install: gyp info spawn args   '-Dnode_engine=v8',
20:05:53  .../node_modules/tree-sitter install: gyp info spawn args   '--depth=.',
20:05:53  .../node_modules/tree-sitter install: gyp info spawn args   '--no-parallel',
20:05:53  .../node_modules/tree-sitter install: gyp info spawn args   '--generator-output',
20:05:53  .../node_modules/tree-sitter install: gyp info spawn args   'build',
20:05:53  .../node_modules/tree-sitter install: gyp info spawn args   '-Goutput_dir=.'
20:05:53  .../node_modules/tree-sitter install: gyp info spawn args ]
20:05:53  .../node_modules/tree-sitter install: gyp info spawn make
20:05:53  .../node_modules/tree-sitter install: gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
20:05:53  .../node_modules/tree-sitter install: make: Entering directory '/home/jenkins/workspace/Geiger/geiger-review/node_modules/.pnpm/tree-sitter@0.20.4/node_modules/tree-sitter/build'
20:05:53  .../node_modules/tree-sitter install:   CC(target) Release/obj.target/tree_sitter/vendor/tree-sitter/lib/src/lib.o
20:05:59  .../node_modules/tree-sitter install: rm -f Release/obj.target/tree_sitter.a Release/obj.target/tree_sitter.a.ar-file-list; mkdir -p `dirname Release/obj.target/tree_sitter.a`
20:05:59  .../node_modules/tree-sitter install: ar crs Release/obj.target/tree_sitter.a @Release/obj.target/tree_sitter.a.ar-file-list
20:05:59  .../node_modules/tree-sitter install:   COPY Release/tree_sitter.a
20:05:59  .../node_modules/tree-sitter install:   CXX(target) Release/obj.target/tree_sitter_runtime_binding/src/binding.o
20:06:00  .../node_modules/tree-sitter install: In file included from ../src/binding.cc:1:
20:06:00  .../node_modules/tree-sitter install: /home/jenkins/.cache/node-gyp/18.16.0/include/node/node.h:997:7: warning: cast between incompatible function types from 'void (*)(v8::Local<v8::Object>)' to 'node::addon_register_func' {aka 'void (*)(v8::Local<v8::Object>, v8::Local<v8::Value>, void*)'} [-Wcast-function-type]
20:06:00  .../node_modules/tree-sitter install:   997 |       (node::addon_register_func) (regfunc),                          \
20:06:00  .../node_modules/tree-sitter install:       |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
20:06:00  .../node_modules/tree-sitter install: /home/jenkins/.cache/node-gyp/18.16.0/include/node/node.h:1031:3: note: in expansion of macro 'NODE_MODULE_X'
20:06:00  .../node_modules/tree-sitter install:  1031 |   NODE_MODULE_X(modname, regfunc, NULL, 0)  // NOLINT (readability/null_usage)
20:06:00  .../node_modules/tree-sitter install:       |   ^~~~~~~~~~~~~
20:06:00  .../node_modules/tree-sitter install: ../src/binding.cc:25:1: note: in expansion of macro 'NODE_MODULE'
20:06:00  .../node_modules/tree-sitter install:    25 | NODE_MODULE(tree_sitter_runtime_binding, InitAll)
20:06:00  .../node_modules/tree-sitter install:       | ^~~~~~~~~~~
20:06:00  .../node_modules/tree-sitter install:   CXX(target) Release/obj.target/tree_sitter_runtime_binding/src/conversions.o
20:06:01  .../node_modules/tree-sitter install:   CXX(target) Release/obj.target/tree_sitter_runtime_binding/src/language.o
20:06:03  .../node_modules/tree-sitter install:   CXX(target) Release/obj.target/tree_sitter_runtime_binding/src/logger.o
20:06:04  .../node_modules/tree-sitter install:   CXX(target) Release/obj.target/tree_sitter_runtime_binding/src/node.o
20:06:06  .../node_modules/tree-sitter install:   CXX(target) Release/obj.target/tree_sitter_runtime_binding/src/parser.o
20:06:07  .../node_modules/tree-sitter install:   CXX(target) Release/obj.target/tree_sitter_runtime_binding/src/query.o
20:06:09  .../node_modules/tree-sitter install:   CXX(target) Release/obj.target/tree_sitter_runtime_binding/src/tree.o
20:06:10  .../node_modules/tree-sitter install:   CXX(target) Release/obj.target/tree_sitter_runtime_binding/src/tree_cursor.o
20:06:12  .../node_modules/tree-sitter install:   CXX(target) Release/obj.target/tree_sitter_runtime_binding/src/util.o
20:06:13  .../node_modules/tree-sitter install:   SOLINK_MODULE(target) Release/obj.target/tree_sitter_runtime_binding.node
20:06:13  .../node_modules/tree-sitter install:   COPY Release/tree_sitter_runtime_binding.node
20:06:13  .../node_modules/tree-sitter install: make: Leaving directory '/home/jenkins/workspace/Geiger/geiger-review/node_modules/.pnpm/tree-sitter@0.20.4/node_modules/tree-sitter/build'
20:06:13  .../node_modules/tree-sitter install: gyp info ok 
20:06:13  .../node_modules/tree-sitter install: Done
```

This is with tree-sitter 0.20.4 and Node 18.16.0 on a Debian CI machine.

--- a/binding.gyp
+++ b/binding.gyp
@@ -43,6 +43,11 @@
             },
           },
         }],
+        ['OS == "linux"', {
+          'cflags_cc': [
+            '-Wno-cast-function-type'
+          ]
+        }],
         ['runtime=="electron"', {
           'defines': ['NODE_RUNTIME_ELECTRON=1']
         }],

--- a/src/binding.cc
+++ b/src/binding.cc
@@ -12,7 +12,7 @@ namespace node_tree_sitter {
 
 using namespace v8;
 
-void InitAll(Local<Object> exports) {
+void InitAll(Local<Object> exports, Local<Value> m_, void* v_) {
   InitConversions(exports);
   node_methods::Init(exports);
   language_methods::Init(exports);

